### PR TITLE
adds api layer for patient service 

### LINKED ISSUE #635: gRPC API layer for patient service

### PR DESCRIPTION
**Pull request by PaulKalho**
## Which issues does this pull request close?
closes #635 

For these endpoints i need projections:
1. GetPatientList
2. GetPatientDetails
3. GetPatientAssignmentByWard
4. GetPatientsByWard
5. GetPatientByBed

I'm also not very happy with the `GetRecentPatients` Endpoint, it seems very inefficient to query each patientAggregate one by one. Can we handle that more efficient with projections?

### CODE REVIEW

**Comment by FoseFx** [comment]
try to run `go mod tidy` in tasks-svc, I think pgerrcode is missing

**Comment by FoseFx** [comment]
I just created an empty tasks_svc database on staging.

Can you update the failing down migrations?

```
echo -n "VERSION=" > $GITHUB_OUTPUT

version_output=$(./migrate.sh $svc version 2>&1 | tail -n1)

# Check if the version_output is a number
if [[ $version_output =~ ^[0-9]+$ ]]; then
  echo $version_output >> $GITHUB_OUTPUT
else
  echo "Error: Version is not a number: $version_output" >&2
  echo "0" >> $GITHUB_OUTPUT 
fi
```

**Review by FoseFx — COMMENTED** [review]
> For these endpoints i need projections

why not implement them in this pr?

**Comment by PaulKalho** [comment]
> why not implement them in this pr?

Postponed them in favor of `property-svc`. We still have some todos in `tasks-svc`, decided to split them up

File: .github/workflows/migrations.yaml [hunk 1]
```diff
@@ -184,12 +184,38 @@ jobs:
       - name: Verify schema.sql and models were generated properly before
         run: git diff --exit-code -- services/${{ matrix.service }} || (echo "You forgot to run ./models.sh before checking in" && exit 1)
       - name: Run DOWN migrations (1/2)
-        run: ./migrate.sh $svc goto $VERSION
+        run: |
+          echo -n "VERSION=" > $GITHUB_OUTPUT
```
[the remark this answers is not in the document]
Comment (FoseFx) [reply .github/workflows/migrations.yaml]: Please undo the changes in the down migration steps, and apply the changes to the version step. (I'm on mobile, so the App could just show me an old version)

**Review by FoseFx — APPROVED** [review]

**Pull request merged**